test/extended/etcd/helpers/helpers.go

```diff
@@ -43,41 +43,50 @@ type TestingT interface {
 	Logf(format string, args ...interface{})
 }
 
-// CreateNewMasterMachine creates a new master node by cloning an existing Machine resource.
-// machineIndex is used to identify the index to which machine should be created to.
-// machineIndex of -1 to be passed if index needn't be taken into consideration while creating the machine
-func CreateNewMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface, machineIndex int64) (string, error) {
+// GetRunningMasterMachine finds and returns a running master machine from the cluster.
+func GetRunningMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (machinev1beta1.Machine, error) {
 	machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
 	if err != nil {
-		return "", err
+		return machinev1beta1.Machine{}, fmt.Errorf("unable to list master machines: %w", err)
 	}
-	var machineToClone *machinev1beta1.Machine
+
 	for _, machine := range machineList.Items {
 		machinePhase := pointer.StringDeref(machine.Status.Phase, "Unknown")
 		if machinePhase == "Running" {
-			machineToClone = &machine
-			break
+			return machine, nil
 		}
 		t.Logf("%q machine is in unexpected %q state", machine.Name, machinePhase)
 	}
+	return machinev1beta1.Machine{}, fmt.Errorf("no running master machines found")
+}
 
-	if machineToClone == nil {
-		return "", fmt.Errorf("unable to find a running master machine to clone")
-	}
-	// assigning a new Name and clearing ProviderID is enough
-	// for MAO to pick it up and provision a new master machine/node
-	if machineIndex == -1 {
+// CreateNewMasterMachine creates a new master node by cloning an existing Machine resource.
+// If machineToClone is provided, it will be cloned. Otherwise a running machine will be picked for cloning.
+func CreateNewMasterMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface, machineToClone machinev1beta1.Machine) (string, error) {
+	//if machineToClone is not provided, get a running master machine to clone
+	if machineToClone.Name == "" {
```

Comment:
I assume this is due to the default/nil of the `machineToClone`? feel free to pass a pointer around, that's a bit easier to digest from a readability PoV